### PR TITLE
DOCSP-30559: compatibility

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-rust/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-30559>
Staging - https://docs-mongodbcom-staging.corp.mongodb.com/rust/docsworker-xlarge/DOCSP-30559-compat/compatibility/

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?

### CODE REVIEW

File: source/includes/language-compatibility-table-rust.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+The MongoDB Rust driver requires Rust 1.57 or later.
```
Comment (mongoKart) [inline source/includes/language-compatibility-table-rust.rst:1]: S: Source constant

Applies throughout

File: source/includes/mongodb-compatibility-table-rust.rst [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+     - ✓
+   * - 2.4 [#2.4-limitation]_
+     - ⊛
+     - ⊛
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+   * - 2.3 [#2.2-limitation]_
```
Comment (mongoKart) [inline source/includes/mongodb-compatibility-table-rust.rst:38]: Q: Confirming this should point to the same footnote as 2.2

File: source/includes/mongodb-compatibility-table-rust.rst [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+   * - 2.0 [#limitations]_
+     - ⊛
+     - ⊛
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+     - ✓
+
+The Rust driver is not compatible with MongoDB server versions older than 3.6.
```
Comment (mongoKart) [inline source/includes/mongodb-compatibility-table-rust.rst:71]:
```suggestion
The Rust driver is not compatible with MongoDB Server versions older than 3.6.
```

File: source/index.txt [hunk 4]
```diff
@@ -78,8 +78,9 @@ Issues & Help
 Compatibility
 -------------
 
-.. For the compatibility charts that show the recommended {+driver-short+} version
-.. for each MongoDB Server version, see :ref:`rust-compatibility`.
+To see the charts that show the recommended {+driver-short+} version
+for each MongoDB Server version and the compatible language versions for
+each driver version, see :ref:`rust-compatibility`.
```
Comment (mongoKart) [inline source/index.txt:83]: I: This sentence is a little hard to parse (and contains the word "version" four times, which might be a record). Removed "recommended" because I don't think that page actually recommends anything.

```suggestion
To see the versions of MongoDB Server and Rust that are compatible with
each version of the {+driver-short+}, see :ref:`rust-compatibility`.
```

Comment (mongoKart) [reply source/index.txt:83]: The above has "see" twice, which I don't love. Could also do this:

```suggestion
For more information about the {+driver-short+}'s compatibility with 
different versions of MongoDB Server and the Rust language, see
:ref:`rust-compatibility`.
```

**Review by mongoKart — CHANGES_REQUESTED** [review]
nice. a couple small things

File: source/includes/mongodb-compatibility-table-rust.rst (hunk 2 above)
Comment (rustagir) [reply source/includes/mongodb-compatibility-table-rust.rst:38]: I'm going to rework these to be reusable.

File: source/index.txt (hunk 4 above)
Comment (rustagir) [reply source/index.txt:83]: good call

File: source/includes/mongodb-compatibility-table-rust.rst [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+.. [#2.5-2.6-limitation] This {+driver-short+} version does not support
+   :manual:`OCSP </core/security-transport-encryption/>`.
+
+.. [#2.4-limitation] This {+driver-short+} version does not support Decimal128
+   or :manual:`OCSP </core/security-transport-encryption/>`.
+
+.. [#2.2-2.3-limitation] This {+driver-short+} version does not support Decimal128,
+   :ref:`Client-Side Field Level Encryption <manual-csfle-feature>`,
+   :manual:`GridFS </core/gridfs/>`, or
+   :manual:`OCSP </core/security-transport-encryption/>`.
```
Comment (mongoKart) [inline source/includes/mongodb-compatibility-table-rust.rst:83]: S: Link straight to OCSP

```suggestion
   :manual:`OCSP </core/security-transport-encryption/#ocsp--online-certificate-status-protocol->`.
```

File: source/includes/mongodb-compatibility-table-rust.rst [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+.. [#2.1-limitation] This {+driver-short+} version does not support Decimal128,
+   :ref:`Client-Side Field Level Encryption <manual-csfle-feature>`,
+   :manual:`GridFS </core/gridfs/>`,
+   :manual:`OCSP </core/security-transport-encryption/>`,
```
Comment (mongoKart) [inline source/includes/mongodb-compatibility-table-rust.rst:88]: S: Link straight to OCSP

```suggestion
   :manual:`OCSP </core/security-transport-encryption/#ocsp--online-certificate-status-protocol->`,
```

File: source/includes/mongodb-compatibility-table-rust.rst [hunk 7]
```diff
@@ -0,0 +1,97 @@
...
+   or :ref:`change streams <changeStreams>`.
+
+.. [#2.0-limitation] This {+driver-short+} version does not support Decimal128,
+   :ref:`Client-Side Field Level Encryption <manual-csfle-feature>`,
+   :manual:`GridFS </core/gridfs/>`,
+   :manual:`OCSP </core/security-transport-encryption/>`,
```
Comment (mongoKart) [inline source/includes/mongodb-compatibility-table-rust.rst:94]: S: Link straight to OCSP

```suggestion
   :manual:`OCSP </core/security-transport-encryption/#ocsp--online-certificate-status-protocol->`,
```

File: source/includes/mongodb-compatibility-table-rust.rst [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+   :ref:`change streams <changeStreams>`,
+   :manual:`Causal Consistency </core/causal-consistency-read-write-concerns>`, or
+   :atlas:`Serverless Instances </reference/serverless-instance-limitations>`.
```
Comment (mongoKart) [inline source/includes/mongodb-compatibility-table-rust.rst:97]: S: Don't link to limitations page (double check the link i suggested)

```suggestion
   :atlas:`Serverless Instances </create-database-deployment/#serverless-instances>`.
```

**Review by mongoKart — APPROVED** [review]
LGTM + some link suggestions!
